core/utils/watchdog.hpp

```diff
@@ -0,0 +1,157 @@
+/**
+ * Copyright Quadrivium LLC
+ * All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <atomic>
+#include <list>
+#include <memory>
+#include <mutex>
+#include <sstream>
+#include <thread>
+#include <unordered_map>
+
+#include <fmt/format.h>
+#include <boost/asio/io_context.hpp>
+
+#ifdef __APPLE__
+
+#include <mach/mach.h>
+#include <mach/thread_act.h>
+#include <mach/thread_info.h>
+
+inline uint64_t getPlatformThreadId() {
+  thread_identifier_info_data_t info;
+  mach_msg_type_number_t size = THREAD_IDENTIFIER_INFO_COUNT;
+  auto r = thread_info(
+      mach_thread_self(), THREAD_IDENTIFIER_INFO, (thread_info_t)&info, &size);
+  if (r != KERN_SUCCESS) {
+    throw std::logic_error{"thread_info"};
+  }
+  return info.thread_id;
+}
+
+#else
+
+#include <sys/syscall.h>
+#include <unistd.h>
+
+inline uint64_t getPlatformThreadId() {
+  return syscall(SYS_gettid);
+}
+
+#endif
+
+namespace kagome {
+
+  constexpr auto kWatchdogDefaultTimeout = std::chrono::minutes{15};
+
+  class Watchdog {
+   public:
+    using Count = uint32_t;
+    using Atomic = std::atomic<Count>;
```

Comment:
```suggestion
    using Atomic = std::atomic_uint32_t;
```

Reply:
`std::atomic<Count>` was used for consistency
("atomic" header says `typedef atomic<uint32_t> atomic_uint32_t;`)